study.md

```diff
@@ -21,18 +21,41 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+<!--
+1.self invoking function expression
+
+https://javascriptweblog.wordpress.com/2010/07/06/function-declarations-vs-function-expressions/
+//**Simulated processing sequence for Question 2**
+function foo(){
+    //a declaration for each function expression
+    var bar = undefined;
+    var bar = undefined;
+    //first Function Expression is executed
+    bar = function() {
+        return 3;
+    };
+    // Function created by first Function Expression is invoked
+    return bar();
+    // second Function Expression unreachable
+}
+alert(foo()); //3
+2.Why is the second function expression unreachable? I understand that a function will exit after it gives a return value but how is the above to return the first and not the second? How is true determined to return the first return statement?-->
```

Comment:
I'm not sure if this answers your question, but I rewrote that code snippet from the page with some messages and hopefully it will help.

```javascript
function foo() {
  console.log('Before the first function expression for bar().');
  console.log('bar\'s value:');
  console.log(bar);
  console.log('-----');


  var bar = function bar1() {
    console.log('First bar\'s body.');

    return 3;
  };

  console.log('After the first function expression for bar().');
  console.log('bar\'s value:');
  console.log(bar);
  console.log('-----');

  console.log('Returning bar().');
  console.log('bar\'s value:');
  console.log(bar);

  return bar();

  console.log('After return bar()');

  var bar = function bar2() {
    console.log('Second bar\'s body');

    return 8;
  };

  console.log('After the second function expression for bar().');
}

console.log('Executing foo().');
console.log('=====');
console.log(foo());

xecuting foo().
=====
Before the first function expression for bar().
bar's value:
undefined
-----
After the first function expression for bar().
bar's value:
[Function: bar1]
-----
Returning bar().
bar's value:
[Function: bar1]
First bar's body.
3
```
Both ```var bar``` statements get hoisted but only the first declaration runs because declarations don't get hoisted, so the second one is never reached.